panther_manager/src/plugins/condition/check_joy_msg.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_manager/plugins/condition/check_joy_msg.hpp"
+
+#include "panther_manager/behavior_tree_utils.hpp"
+
+namespace panther_manager
+{
+
+BT::NodeStatus CheckJoyMsg::onTick(const JoyMsg::SharedPtr & last_msg)
+{
+  return (last_msg && checkAxes(last_msg) && checkButtons(last_msg) && checkTimeout(last_msg))
+           ? BT::NodeStatus::SUCCESS
+           : BT::NodeStatus::FAILURE;
+}
+
+bool CheckJoyMsg::checkAxes(const JoyMsg::SharedPtr & last_msg)
+{
+  std::vector<float> expected_axes;
+  getInput<std::vector<float>>("axes", expected_axes);
+
+  if (expected_axes.empty()) {
+    return true;
+  }
+
+  if (last_msg->axes.size() != expected_axes.size()) {
+    RCLCPP_WARN_STREAM(
+      this->logger(), GetLoggerPrefix(name())
+                        << "Joy message has " << last_msg->axes.size()
+                        << " axes, expected at least " << expected_axes.size());
+    return false;
+  }
+
+  return std::equal(expected_axes.begin(), expected_axes.end(), last_msg->axes.begin());
```

Comment:
One sensible use is to check if, for example, the speed value is 0. In some cases this can be helpful. However, in most cases this field will most likely be empty.